packages/ai-api/src/utils/deployment-resolver.ts

```diff
@@ -48,6 +48,17 @@ export type ModelDeployment<ModelNameT = string> =
   | ModelNameT
   | ((ModelConfig<ModelNameT> | DeploymentIdConfig) & ResourceGroupConfig);
 
+/**
+ * @internal
+ */
+export function getResourceGroup(
+  modelDeployment: ModelDeployment
+): string | undefined {
+  return typeof modelDeployment === 'object'
+    ? modelDeployment.resourceGroup
+    : undefined;
```

Comment:
(minor) Consider: 

```suggestion
): string {
  return typeof modelDeployment === 'object'
    ? modelDeployment.resourceGroup
    : 'default' ;
```

Because if I see it correctly in all usages `undefined` is then actually turned into the default RG.

Reply:
undefined is never assigned with the `??` operator, e.g.
```
resourceGroup ?? 'default',
```
I would keep the function as is, as the model deployment doesn't contain anything.

Reply:
Otherwise the function would be called `getResourceGroupWithDefault` or so

Reply:
Then why not just call it that way? Because otherwise we are duplicating `?? 'default'` in every usage. Assuming there are multiple usages, otherwise no point in having a getter, right?

Reply:
I would argue the purpose of the getter is to handle the different types the property can hold, e.g. string or object.

Reply:
I will change the function to WithDefault, with an optional parameter holding a different default than 'default'

Reply:
On second thought, this doesn't improve the situation, as I would then need to change the endpoint type to require this property, which would then change every call of executeRequest.
Also we only do the `??` logic once in the actual function and once in a mock, so it doesn't really provide a lot of value.

-> I suggest not changing it